substrate/client/chain-spec/src/chain_spec.rs

```diff
@@ -42,17 +42,31 @@ use std::{
 #[derive(Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 enum GenesisBuildAction<EHF> {
+	/// Gets the default config (aka PresetId=None) and patches it.
 	Patch(json::Value),
+	/// Assumes that the full `RuntimeGenesisConfig` is supplied.
 	Full(json::Value),
-	NamedPreset(String, PhantomData<EHF>),
+	/// Gets the named preset and applies an optional patch.
+	NamedPreset(String, json::Value, PhantomData<EHF>),
```

Comment:
Not sure if that would make some more sense:
```suggestion
	NamedPreset(String, Option<json::Value>, PhantomData<EHF>),
```

Reply:
I was weighing the options here, and opted against it, as this allows us to treat all 3 variants the same way.

Reply:
Option should not bring that much mess in handling it, but yeah, it is also acceptable with empty json.